### PR TITLE
Small git utils improvements

### PR DESCRIPTION
**Pull request by mbtools**
- Saves ~20ms per call of `length_uft8_hex`
- Improve test coverage

### CODE REVIEW

File: src/git/zcl_abapgit_git_utils.clas.testclasses.abap [hunk 1]
```diff
@@ -1,32 +1,77 @@
...
+
+    CONSTANTS lc_null TYPE x LENGTH 2 VALUE '0000'.
+
+    DATA:
+      lv_c TYPE c LENGTH 1,
+      lv_x TYPE x LENGTH 2.
+
+    lv_c = mo_cut->get_null( ).
+
+    lv_x = lv_c.
```
Comment (larshp) [inline src/git/zcl_abapgit_git_utils.clas.testclasses.abap:32]: this doesnt test the right thing? moving any non hex value from char to hex will give the initial hex value, 

```abap
DATA lv_c TYPE c LENGTH 1.
DATA lv_x TYPE x LENGTH 2.
lv_c = 'Q'.
lv_x = lv_c.
WRITE lv_x.
```

**Comment by larshp** [comment]
heh, yea, so thats probably another transpiler bug, CASTING char to hex, whereas hex to char works

**Comment by mbtools** [comment]
yeah pushing the limits 😀 

```abap
FIELD-SYMBOLS:
  <i> TYPE i,
  <c> TYPE c,
  <x> TYPE x,
  <s> TYPE string,
  <any> TYPE any.
```

all of these transpile to the same `let fs_var_ = new abap.types.FieldSymbol();`. so the type isn't passed in.

**Comment by larshp** [comment]
its now passed in 😁 https://github.com/abaplint/transpiler/pull/748/commits/ef63a46f8f4b8ef769cdb77f661b86f46cb0dedb

**Review by larshp — APPROVED** [review]

**Comment by mbtools** [comment]
alright 👍

**Pull request merged**